slick-testkit/src/main/scala/com/typesafe/slick/testkit/tests/InsertTest.scala

```diff
@@ -54,6 +54,23 @@ class InsertTest extends AsyncTest[JdbcTestDB] {
     )
   }
 
+  def testUniqueInsert = {
+    case class ARow(email: String , id: Int = 0)
+    class A(tag: Tag) extends Table[ARow](tag , "A_UNIQUEINSERT"){
+      def id = column[Int]("id" , O.AutoInc , O.PrimaryKey)
+      def email = column[String]("email" , O.Unique)
+
+      def * = (email , id)<>(ARow.tupled , ARow.unapply )
+    }
+    val atq = TableQuery[A]
+
+    db.run(atq.schema.create)
```

Comment:
Seq( atq.schema.create,
atq ++= Seq( ARow("unique@site.com") , ARow("user@site.com") )) ,
atq += ARow("unique@site.com") //test this failure
)